File: springapp/src/main/java/com/examly/springapp/service/InvestmentService.java
package com.examly.springapp.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.examly.springapp.model.Investment;

@Service
public interface InvestmentService {

    public Investment addInvestment(Investment investment);
    public Investment updateInvestment(long investmentId, Investment updatedInvestment);
    public Investment getInvestmentById(long investmentId);
    public List<Investment> getAllInvestments();
    public List<Investment> getInvestmentsByType(String type);
    public List<Investment> getInvestmentsByStatus(String status);
    public List<Investment> searchInvestments(String keyword);
    public void deleteInvestment(Long investmentId);
    
}

File: angularapp/src/app/services/investment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Investment } from '../models/investment.model';
import { AppConstant } from '../app.constant';
@Injectable({
  providedIn: 'root'
})
export class InvestmentService {

  private apiUrl = AppConstant.apiUrl;

  getAllInvestments():Observable<any>{
    return this.client.get(`${this.apiUrl}/investments`);
  }
 
  getInvestmentById(investmentId:number):Observable<any>{
    return this.client.get(`${this.apiUrl}/investments/${investmentId}`);
  }
 
  addInvestment(investment:Investment):Observable<any>{
    return this.client.post(`${this.apiUrl}/investments`,investment);
  }
 
  updateInvestment(investmentId:number,investment:Investment):Observable<any>{
    return this.client.put(`${this.apiUrl}/investments/${investmentId}`,investment);
  }
 
  deleteInvestment(investmentId:number):Observable<any>{
    return this.client.delete(`${this.apiUrl}/investments/${investmentId}`);
  }
 
 
  constructor(private client:HttpClient) { }
}
